plugins/home/src/api/VisitsApiFactory.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Visit,
+  VisitsApi,
+  VisitsApiQueryParams,
+  VisitsApiSaveParams,
+} from './VisitsApi';
+
+type ArrayElement<A> = A extends readonly (infer T)[] ? T : never;
+
+/** @public */
+export type VisitsApiFactoryOptions = {
+  limit: number;
+  retrieveAll?: () => Promise<Array<Visit>>;
+  persistAll?: (visits: Array<Visit>) => Promise<void>;
+};
+
+/**
+ * @public
+ * This helps the creation of VisitApi implementations. Important to note
+ * that it implements features like orderBy and filterBy on memory, therefore
+ * is intended to handle few visits. The default is 100.
+ * See LocalStorageVisitsApi for an usage example.
+ */
+export class VisitsApiFactory implements VisitsApi {
+  protected readonly randomUUID = window.crypto.randomUUID;
+  protected readonly limit: number;
```

Comment:
Suggestion (non-blocking):
This is great to be able to customize how `id` is generated, and makes sense to me to be customized by the developer providing the implementation, so via code callback.
But for the limit itself, I think it is more an app integrator's responsibility, so I would suggest getting the limit from the app config instead.

Reply:
That is a good idea. I will save it for a followup PR.